fintech-examples/fintech-ui/src/app/services/storage.service.ts

```diff
@@ -52,16 +53,26 @@ export class StorageService {
   public setMaxAge(maxAge: number): void {
     const timestamp = new Date().getTime() + maxAge * 1000;
     localStorage.setItem(Session.MAX_VALID_UNTIL, '' + timestamp);
-    console.log('set max age ' + maxAge + ' till ' + new Date(timestamp).toLocaleString());
+    console.log('set max age ' + maxAge + ' till ' + toLocaleString(new Date(timestamp)));
+  }
+
+  public setRedirectCode(redirectCode: string): void {
+    localStorage.setItem(Session.REDIRECT_CODE, redirectCode);
+  }
+
+  public getRedirectCode(): string {
+    return localStorage.getItem(Session.REDIRECT_CODE);
   }
 
   public getValidUntilDate(): Date {
     const validUntilTimestamp = localStorage.getItem(Session.MAX_VALID_UNTIL);
     if (validUntilTimestamp === undefined || validUntilTimestamp === null) {
       return null;
     }
-    const date =  new Date(parseInt(validUntilTimestamp, 10));
-    if (date.toLocaleString() === 'Invalid Date') {
+    const date = new Date(parseInt(validUntilTimestamp, 0));
+    if (toLocaleString(date) === 'Invalid Date') {
```

Comment:
Using this hardcoded literal is really bad. I would advise using date library like moment or luxon for date operations - and I see there are lots of date operations